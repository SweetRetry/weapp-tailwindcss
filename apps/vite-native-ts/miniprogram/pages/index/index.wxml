<view class="p-4">
  <IceButton class="bg-[#d72929]" my-class="bg-[orange]"></IceButton>
</view>