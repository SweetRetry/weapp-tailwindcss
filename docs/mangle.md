# mangle 配置

用于压缩混淆所有的 `className`，类似:

- `flex` -> `a`
- `w-[200rpx]` -> `b`
- `text-[rgba(255,255,0,1)]` -> `c`

当然，你可以自定义混淆生成的所有 `className`

> 当前版本，只会对所有静态节点上的  `className` 进行替换，如果一个节点通过 `js` 来绑定`class`，则这个节点上，所有的 `class` 都不会被混淆替换。

Typescript 类型为

```ts

const mangle : boolean | IMangleOptions

export interface IMangleOptions {
  /**
   * 保留指定的className
   */
  reserveClassName?: (string | RegExp)[]
  /**
   * 自定义className生成器
   */
  classGenerator?: (original: string, opts: IMangleOptions, context: Record<string, any>) => string | undefined
  /**
   * 开启console.log打印用于调试
   */
  log?: boolean
  /**
   * 排除指定路径下的文件，优先级高于 include
   */
  exclude?: (string | RegExp)[]
  /**
   * 包括指定路径下的文件，优先级低于 exclude
   */
  include?: (string | RegExp)[]
  /**
   * 忽略指定的className
   */
  ignoreClass?: (string | RegExp)[]
}
```

当 `mangle` 为 `false` 为不开启此功能

当 `mangle` 为 `true`，默认生效的配置项为：

```ts
{
  exclude: [/node[-_]modules/, /(wx|my|swan|tt|ks|jd)components/]
}
```

此配置项意义为，默认不处理第三方UI库中的所有 `className`。

如果你想保留 `css` 选择器不让被替换，可以使用 `/* mangle disabled */` 注释或者 `/* mangle ignore */` 注释，效果是一样的：

```css
/* mangle disabled */
.el-tree-node__content {
  color: red;
}
```
