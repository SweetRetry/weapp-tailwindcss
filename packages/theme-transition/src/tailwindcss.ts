import plugin from 'tailwindcss/plugin'

export const themeTransition = plugin.withOptions(() => {
  return ({ addBase }) => {
    addBase({

    })
  }
})
