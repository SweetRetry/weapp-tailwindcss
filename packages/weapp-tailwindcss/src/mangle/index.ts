export * from '@weapp-tailwindcss/mangle'
