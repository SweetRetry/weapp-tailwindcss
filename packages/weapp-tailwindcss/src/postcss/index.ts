export * from '@weapp-tailwindcss/postcss'
