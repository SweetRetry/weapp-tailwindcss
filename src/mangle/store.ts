import { ClassGenerator, defaultMangleClassFilter } from 'tailwindcss-mangle-shared'
import { UserDefinedOptions } from '@/types'
import { splitCode } from '@/extractors/split'
import { escapeStringRegexp } from '@/reg'

function getSelf(x: string) {
  return x
}

export interface IMangleScopeStore {
  rawOptions: UserDefinedOptions['mangle']
  runtimeSet: Set<string>
  classGenerator: ClassGenerator
  filter: (className: string) => boolean
  cssHandler: (rawSource: string) => string
  jsHandler: (rawSource: string) => string
  wxmlHandler: (rawSource: string) => string
}

const defaultScopedStore: IMangleScopeStore = {
  rawOptions: false,
  runtimeSet: new Set<string>(),
  classGenerator: new ClassGenerator(),
  filter: defaultMangleClassFilter,
  cssHandler: getSelf,
  jsHandler: getSelf,
  wxmlHandler: getSelf
} // as default
// import type { BabelFileResult } from '@babel/core'
// 为什么要用这种方式，因为一层一层往下传递参数，太烦了
// 所以说还是搞个 IOC 容器比较爽，要什么注入什么
// 这样写其实是很有问题的
// 比如当插件被 new 了 2次，传入了不同的 store 然后就挂了
// TODO
const store = Object.assign({}, defaultScopedStore)

export function useStore() {
  return store
}

export function resetStore() {
  return Object.assign(store, defaultScopedStore)
}

function handleValue(rawSource: string) {
  const arr = splitCode(rawSource)
  for (const x of arr) {
    if (store.runtimeSet.has(x)) {
      rawSource = rawSource.replaceAll(new RegExp(escapeStringRegexp(x), 'g'), store.classGenerator.generateClassName(x).name)
    }
  }
  return rawSource
}

export function initStore(options: UserDefinedOptions['mangle']) {
  store.rawOptions = options

  if (options) {
    if (options === true) {
      options = {
        classGenerator: {},
        mangleClassFilter: defaultMangleClassFilter
      }
    }
    store.classGenerator = new ClassGenerator(options.classGenerator)
    store.filter = options.mangleClassFilter ?? defaultMangleClassFilter
    store.jsHandler = (rawSource: string) => {
      return handleValue(rawSource)
    }
    store.cssHandler = (rawSource: string) => {
      // process 最后处理, loader 无所谓顺序
      return handleValue(rawSource)
    }

    store.wxmlHandler = (rawSource: string) => {
      //  splitCode(rawSource)
      return handleValue(rawSource)
    }
  }
}

export function setRuntimeSet(runtimeSet: Set<string>) {
  const newSet = new Set<string>()
  for (const c of runtimeSet) {
    // eslint-disable-next-line unicorn/no-array-callback-reference
    if (store.filter(c)) {
      newSet.add(c)
    }
  }
  store.runtimeSet = newSet
}
