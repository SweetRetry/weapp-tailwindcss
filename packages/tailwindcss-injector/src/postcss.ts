import type { PluginCreator } from 'postcss'
import type { Config } from 'tailwindcss'
import { defu } from 'defu'
import postcss from 'postcss'
// import set from 'set-value'
import tailwindcss from 'tailwindcss'

// function isObject(obj: any): obj is object {
//   return typeof obj === 'object' && obj !== null
// }

// function isTailwindcssConfig(obj: any): obj is Config {
//   return isObject(obj)
// }

type InlineTailwindcssOptions = string | Config | {
  config: string | Config
} | undefined

export interface Options {
  filter: (input?: postcss.Input) => boolean
  config: InlineTailwindcssOptions | ((input?: postcss.Input) => InlineTailwindcssOptions)
  directiveParams: ('base' | 'components' | 'utilities' | 'variants')[]
}

const creator: PluginCreator<Partial<Options>> = (options) => {
  const { config, filter, directiveParams } = defu<Options, Options[]>(options, {
    filter: () => true,
    config: undefined,
    directiveParams: ['utilities'],
  })
  return {
    postcssPlugin: 'postcss-tailwindcss-injector',
    async Once(root, helpers) {
      if (filter(root.source?.input)) {
        let _root = root
        for (const params of directiveParams) {
          const node = root.nodes.find(x => x.type === 'atrule' && x.params === params)
          if (!node) {
            _root = _root.prepend(helpers.atRule({ name: 'tailwind', params }))
          }
        }

        const cfg = typeof config === 'function' ? config(root.source?.input) : config
        // if (isTailwindcssConfig(cfg)) {
        //   if ('config' in cfg) {

        //   }
        //   else {
        //     cfg.content = []
        //   }
        // }
        const { root: newRoot } = await postcss([
          tailwindcss(cfg),
        ]).process(_root)

        root.replaceWith(newRoot)
      }
    },
  }
}
creator.postcss = true

export default creator
