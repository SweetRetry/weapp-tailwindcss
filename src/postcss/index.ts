import postcss from 'postcss'

import { mpRulePreflight, commonChunkPreflight } from './mp'
// mpAtRulePreflight
const isMp = true
export function styleHandler (rawSource: string, file: string) {
  const root = postcss.parse(rawSource)
  // console.log(file)
  root.walk((node, idx) => {
    if (node.type === 'rule') {
      if (isMp) {
        // 引用传递
        // uni-app common-> main.wxss
        // taro app.wxss
        if (file === 'main' || file === 'app') {
          commonChunkPreflight(node)
        }
        mpRulePreflight(node)
      }
    }
    //  else if (node.type === 'atrule') {
    //   if (isMp) {
    //     mpAtRulePreflight(node)
    //   }
    // }
    // 顺其自然
    // else if (node.type === 'comment') {
    //   node.remove()
    // }
  })
  return root.toString()
}
