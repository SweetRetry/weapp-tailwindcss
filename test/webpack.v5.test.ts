import type { Compiler } from 'webpack'
import { UnifiedWebpackPluginV5 } from '@/index'
import { getCompiler5, compile, readAssets, createLoader, getErrors, getWarnings } from './helpers'
import path from 'path'
import postcss from 'postcss'
import fs from 'fs/promises'
import { useStore } from '@/mangle/store'

describe('webpack5 plugin', () => {
  let compiler: Compiler
  let prodCompiler: Compiler
  beforeEach(() => {
    const processor = postcss([
      // require('autoprefixer')(),
      require('tailwindcss')({ config: path.resolve(__dirname, './fixtures/webpack/v5/wxml/tailwind.config.js') })
      // require('postcss-rem-to-responsive-pixel')({
      //   rootValue: 32,
      //   propList: ['*'],
      //   transformUnit: 'rpx'
      // })
    ])
    compiler = getCompiler5({
      mode: 'development',
      entry: {
        wxml: path.resolve(__dirname, './fixtures/webpack/v5/wxml/pages/index.js')
      },
      output: {
        path: path.resolve(__dirname, './dist'),
        filename: '[name].js', // ?var=[fullhash]
        chunkFilename: '[id].[name].js' // ?ver=[fullhash]
      },
      module: {
        rules: [
          {
            test: /\.m?js$/,
            // https://webpack.js.org/configuration/module/#useentry
            use: createLoader(async function (source) {
              const basename = path.basename(this.resourcePath, path.extname(this.resourcePath))
              const filename = basename + '.wxml'
              const content = await fs.readFile(path.resolve(this.context, filename), {
                encoding: 'utf-8'
              })
              this.emitFile(filename, content)

              const cssFilename = basename + '.css'

              const cssContent = await fs.readFile(path.resolve(this.context, cssFilename), {
                encoding: 'utf-8'
              })

              const res = await processor.process(cssContent, {
                from: undefined,
                map: false
              })
              this.emitFile(cssFilename, res.toString())
              return source
              // this.emitFile('hello.xx', '12345')
              // console.log()
            })
          }
          // {
          //   test: /\.css$/,
          //   use: [
          //     createLoader(function (source) {
          //       const basename = path.basename(this.resourcePath, path.extname(this.resourcePath))
          //       const filename = basename + '.css'
          //       this.emitFile(filename, source)
          //       return ''
          //     }),
          //     {
          //       loader: 'postcss-loader',
          //       options: {
          //         postcssOptions: {
          //           plugins: [require('tailwindcss')({ config: path.resolve(__dirname, './fixtures/webpack/v5/wxml/tailwind.config.js') })]
          //         }
          //       }
          //     }
          //   ]
          // },
          // {
          //   test: /\.wxml$/,
          //   use: createLoader(function (source) {
          //     const basename = path.basename(this.resourcePath, path.extname(this.resourcePath))
          //     const filename = basename + '.wxml'
          //     this.emitFile(filename, source)
          //   })
          // }
        ]
      }
    })

    prodCompiler = getCompiler5({
      mode: 'production',
      entry: {
        wxml: path.resolve(__dirname, './fixtures/webpack/v5/wxml/pages/index.js')
      },
      output: {
        path: path.resolve(__dirname, './dist'),
        filename: '[name].js', // ?var=[fullhash]
        chunkFilename: '[id].[name].js' // ?ver=[fullhash]
      },
      module: {
        rules: [
          {
            test: /\.m?js$/,
            use: createLoader(async function (source) {
              const basename = path.basename(this.resourcePath, path.extname(this.resourcePath))
              const filename = basename + '.wxml'
              const content = await fs.readFile(path.resolve(this.context, filename), {
                encoding: 'utf-8'
              })
              this.emitFile(filename, content)

              const cssFilename = basename + '.css'

              const cssContent = await fs.readFile(path.resolve(this.context, cssFilename), {
                encoding: 'utf-8'
              })

              const res = await processor.process(cssContent, {
                from: undefined,
                map: false
              })
              this.emitFile(cssFilename, res.toString())
              return source
            })
          }
        ]
      }
    })
  })
  it('common', async () => {
    new UnifiedWebpackPluginV5({
      mainCssChunkMatcher(name) {
        return path.basename(name) === 'index.css'
      },
      customReplaceDictionary: 'complex'
    }).apply(compiler)

    const stats = await compile(compiler)

    expect(readAssets(compiler, stats)).toMatchSnapshot('assets')
    expect(getErrors(stats)).toMatchSnapshot('errors')
    expect(getWarnings(stats)).toMatchSnapshot('warnings')
  })

  it('unified common', async () => {
    new UnifiedWebpackPluginV5({
      mainCssChunkMatcher(name) {
        return path.basename(name) === 'index.css'
      }
    }).apply(compiler)

    const stats = await compile(compiler)

    expect(readAssets(compiler, stats)).toMatchSnapshot('assets')
    expect(getErrors(stats)).toMatchSnapshot('errors')
    expect(getWarnings(stats)).toMatchSnapshot('warnings')
  })

  it('unified prod common', async () => {
    new UnifiedWebpackPluginV5({
      mainCssChunkMatcher(name) {
        return path.basename(name) === 'index.css'
      }
    }).apply(prodCompiler)

    const stats = await compile(prodCompiler)

    expect(readAssets(prodCompiler, stats)).toMatchSnapshot('assets')
    expect(getErrors(stats)).toMatchSnapshot('errors')
    expect(getWarnings(stats)).toMatchSnapshot('warnings')
  })

  it('disabled true', async () => {
    new UnifiedWebpackPluginV5({
      mainCssChunkMatcher(name) {
        return path.basename(name) === 'index.css'
      },
      disabled: true
    }).apply(compiler)

    const stats = await compile(compiler)

    expect(readAssets(compiler, stats)).toMatchSnapshot('assets')
    expect(getErrors(stats)).toMatchSnapshot('errors')
    expect(getWarnings(stats)).toMatchSnapshot('warnings')
  })

  it('mangle true', async () => {
    new UnifiedWebpackPluginV5({
      mainCssChunkMatcher(name) {
        return path.basename(name) === 'index.css'
      },

      customReplaceDictionary: 'complex',
      mangle: true
    }).apply(compiler)

    const stats = await compile(compiler)
    const { runtimeSet, classGenerator, recorder } = useStore()
    expect(runtimeSet.size).toBeGreaterThan(0)
    expect(recorder.js.length).toBeGreaterThan(0)
    expect(recorder.css.length).toBeGreaterThan(0)
    expect(recorder.wxml.length).toBeGreaterThan(0)
    expect(classGenerator.newClassSize).toBeGreaterThan(0)
    expect(readAssets(compiler, stats)).toMatchSnapshot('assets')
    expect(getErrors(stats)).toMatchSnapshot('errors')
    expect(getWarnings(stats)).toMatchSnapshot('warnings')
  })

  it('mangle options', async () => {
    new UnifiedWebpackPluginV5({
      mainCssChunkMatcher(name) {
        return path.basename(name) === 'index.css'
      },
      mangle: {
        classGenerator: {
          classPrefix: ''
        }
      },
      customReplaceDictionary: 'complex'
    }).apply(compiler)

    const stats = await compile(compiler)
    const { runtimeSet, classGenerator, recorder } = useStore()
    expect(runtimeSet.size).toBeGreaterThan(0)
    expect(recorder.js.length).toBeGreaterThan(0)
    expect(recorder.css.length).toBeGreaterThan(0)
    expect(recorder.wxml.length).toBeGreaterThan(0)
    expect(classGenerator.newClassSize).toBeGreaterThan(0)
    expect(readAssets(compiler, stats)).toMatchSnapshot('assets')
    expect(getErrors(stats)).toMatchSnapshot('errors')
    expect(getWarnings(stats)).toMatchSnapshot('warnings')
  })
})
