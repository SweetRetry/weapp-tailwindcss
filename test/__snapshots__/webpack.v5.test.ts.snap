// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`webpack5 plugin common: assets 1`] = `
Object {
  "index.css": "view, ::before, ::after {
    --tw-translate-x: 0;
    --tw-translate-y: 0;
    --tw-rotate: 0;
    --tw-skew-x: 0;
    --tw-skew-y: 0;
    --tw-scale-x: 1;
    --tw-scale-y: 1;
    --tw-pan-x:  ;
    --tw-pan-y:  ;
    --tw-pinch-zoom:  ;
    --tw-scroll-snap-strictness: proximity;
    --tw-ordinal:  ;
    --tw-slashed-zero:  ;
    --tw-numeric-figure:  ;
    --tw-numeric-spacing:  ;
    --tw-numeric-fraction:  ;
    --tw-ring-inset:  ;
    --tw-ring-offset-width: 0px;
    --tw-ring-offset-color: #fff;
    --tw-ring-color: rgb(59 130 246 / 0.5);
    --tw-ring-offset-shadow: 0 0 #0000;
    --tw-ring-shadow: 0 0 #0000;
    --tw-shadow: 0 0 #0000;
    --tw-shadow-colored: 0 0 #0000;
    --tw-blur:  ;
    --tw-brightness:  ;
    --tw-contrast:  ;
    --tw-grayscale:  ;
    --tw-hue-rotate:  ;
    --tw-invert:  ;
    --tw-saturate:  ;
    --tw-sepia:  ;
    --tw-drop-shadow:  ;
    --tw-backdrop-blur:  ;
    --tw-backdrop-brightness:  ;
    --tw-backdrop-contrast:  ;
    --tw-backdrop-grayscale:  ;
    --tw-backdrop-hue-rotate:  ;
    --tw-backdrop-invert:  ;
    --tw-backdrop-opacity:  ;
    --tw-backdrop-saturate:  ;
    --tw-backdrop-sepia:  
;
    box-sizing: border-box;
    border-width: 0;
    border-style: solid;
    border-color: currentColor}",
  "index.wxml": "<!-- index.wxml -->
<import src=\\"../../wxml/motto/motto.wxml\\" />
<view class=\\"container\\">
  <view bindtap=\\"bindViewTap\\">
    <user avatarUrl=\\"{{userInfo.avatarUrl}}\\" nickName=\\"{{userInfo.nickName}}\\"></user>
  </view>
  <view class=\\"motto\\">
    <template is=\\"motto\\" data=\\"{{motto:motto}}\\" />
  </view>
  <view class=\\"powered h-_l_200_pct__r_\\">
    <text>{{powered.by}}</text>
  </view>
  <view class=\\"-mt-2 mb-_l_-20px_r_ p-_l_20px_r_\\">
    p-[20px] -mt-2 mb-[-20px] margin的jit 可不能这么写 -m-[20px]
  </view>
  <view class=\\"space-y-_l_1-dot-6rem_r_\\">
    <view class=\\"w-_l_300rpx_r_ text-black text-opacity-_l_0-dot-19_r_\\">
      w-[300rpx] text-black text-opacity-[0.19]
    </view>
    <view class=\\"max-h-_l_100px_r_ min-w-_l_300rpx_r_ text-_l_20px_r_ leading-_l_0-dot-9_r_\\">
      min-w-[300rpx] max-h-[100px] text-[20px] leading-[0.9]
    </view>
    <view class=\\"min-h-_l_100px_r_ max-w-_l_300rpx_r_ text-_l__h_dddddd_r_\\">
      max-w-[300rpx] min-h-[100px] text-[#dddddd]
    </view>
    <view class=\\"flex h-_l_100px_r_ w-_l_100px_r_ items-center justify-center rounded-_l_40px_r_ bg-_l__h_123456_r_ bg-opacity-_l_0-dot-54_r_ text-_l__h_ffffff_r_\\">
      Hello
    </view>
    <view class=\\"border-_l_10px_r_ border-solid border-_l__h_098765_r_ border-opacity-_l_0-dot-44_r_\\">
      border-[10px] border-[#098765] border-solid border-opacity-[0.44]
    </view>
    <view class=\\"grid grid-cols-3 divide-x-_l_10px_r_ divide-solid divide-_l__h_010101_r_\\">
      <div>1</div>
      <div>2</div>
      <div>3</div>
    </view>
    <view class=\\"w-32 rounded-md bg-pink-500 py-2 font-semibold text-white ring-4 ring-pink-300\\">
      Default
    </view>
  </view>
  <view class=\\"test\\">test</view>
  <view class=\\"border-0 icon h-10 w-10 mx-auto {{active==\\"home\\"?\\"icon-home-selected\\":\\"icon-home\\"}} {{b}} {{a===\\"cc\\"}} w-_l_20px_r_ {{flag==\\"p-_l_20px_r_\\"?\\"p-_l_20px_r_\\":\\"m-_l_20px_r_\\"}}\\"></view>
  <button class=\\"bg-sky-700 px-4 py-2 text-white hover_c_bg-sky-800 sm_c_px-8 sm_c_py-3\\">...</button>
  <view class=\\"border-0 icon h-10 w-10 mx-auto {{active==\\"home\\"?\\"icon-home-selected\\":\\"icon-home\\"}} {{b}} {{a===\\"cc\\"}} w-_l_20px_r_ {{flag==\\"p-_l_20px_r_\\"?\\"p-_l_20px_r_\\":\\"m-_l_20px_r_\\"}}\\"></view>
</view>",
  "wxml.js?var=87de820fb9efeaeb2cc6": "/*
 * ATTENTION: The \\"eval\\" devtool has been used (maybe by default in mode: \\"development\\").
 * This devtool is neither made for production nor for readable output files.
 * It uses \\"eval()\\" calls to create a separate source file in the browser devtools.
 * If you are trying to read the output file, select a different devtool (https://webpack.js.org/configuration/devtool/)
 * or disable the default devtool with \\"devtool: false\\".
 * If you are looking for production-ready output files, see mode: \\"production\\" (https://webpack.js.org/configuration/mode/).
 */
/******/ (() => { // webpackBootstrap
/******/ 	var __webpack_modules__ = ({

/***/ \\"./test/fixtures/webpack/v5/wxml/pages/index.js\\":
/*!******************************************************!*\\\\
  !*** ./test/fixtures/webpack/v5/wxml/pages/index.js ***!
  \\\\******************************************************/
/***/ (() => {

eval(\\"\\\\n\\\\n//# sourceURL=webpack://weapp-tailwindcss-webpack-plugin/./test/fixtures/webpack/v5/wxml/pages/index.js?\\");

/***/ })

/******/ 	});
/************************************************************************/
/******/ 	
/******/ 	// startup
/******/ 	// Load entry module and return exports
/******/ 	// This entry module can't be inlined because the eval devtool is used.
/******/ 	var __webpack_exports__ = {};
/******/ 	__webpack_modules__[\\"./test/fixtures/webpack/v5/wxml/pages/index.js\\"]();
/******/ 	
/******/ })()
;",
}
`;

exports[`webpack5 plugin common: errors 1`] = `Array []`;

exports[`webpack5 plugin common: warnings 1`] = `Array []`;
